os/kernel/group/group_create.c

```diff
@@ -321,7 +324,13 @@ int group_initialize(FAR struct task_tcb_s *tcb)
 	irqrestore(flags);
 
 #endif
+#if defined(CONFIG_PREFERENCE) && CONFIG_TASK_NAME_SIZE > 0
+	/* The values of private preference are managed by task group.
+	 * Set a name of task group, tg_name for private preference. */
+
+	strncpy(group->tg_name, tcb->cmn.name, CONFIG_TASK_NAME_SIZE);
```

Comment:
Is it okay to use thread name as preference identifier? What I worried is thread name can be duplicated.

Reply:
I agree with your concerns. But a name is only identifier to identify task group I think.. Do you have any ideas for this?

Reply:
How about appending the group id or task pid for identifying?

Reply:
A group id or pid would be different after reloading or reboot.